### PR TITLE
Add crypto crate to the toolkit

### PR DESCRIPTION
**Pull request by PumpkinSeed**
Based on the [I609](https://github.com/enigmampc/SecretNetwork/issues/609) I moved the original content to the crypto crate and make them public based on the pattern of the other crates.

Also changes on my side: 
- Added the verify function to the secp256k1.
- Add tests for all of the public functions. 

Leftover:
I didn't add **ed25519** signing however the original ticket contains it. This requires a decision on your part, since I found a [crate](https://docs.rs/ed25519/1.0.3/ed25519/) for that purpose, I would say do it in a different PR. 
Also to make it consistent, currently the sign.rs only support secp256k1, I don't know whether we can use the same parameter-set for ed25529, but if yes we can create an enum for decide between algorithms to sign the data. 

```
enum SignAlgorithm {
  Secp256k1
  Ed25519
}
```

### CODE REVIEW

**Comment by Cashmaney** [comment]
Thanks @PumpkinSeed! @reuvenpo can you take a look and make sure everything is rusty?

File: packages/crypto/src/sign.rs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    #[test]
+    fn test_pubkey() {
+        let s = Secp256k1::new();
+        let (secp_privkey, secp_pubkey) = s.generate_keypair(&mut thread_rng());
+
+        let mut privkey = [0u8; 32];
+        for i in 0..32 {
+            privkey[i] = secp_privkey[i];
+        }
```
Comment (reuvenpo) [inline packages/crypto/src/sign.rs:36]: you can probably use `copy_from_slice` here instead of this `for` loop

File: packages/crypto/src/sign.rs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    #[test]
+    fn test_sign() {
+        let s = Secp256k1::new();
+        let (secp_privkey, secp_pubkey) = s.generate_keypair(&mut thread_rng());
+
+        let mut privkey = [0u8; 32];
+        for i in 0..32 {
+            privkey[i] = secp_privkey[i];
+        }
```
Comment (reuvenpo) [inline packages/crypto/src/sign.rs:55]: you can probably use `copy_from_slice` here instead of this `for` loop

**Review by reuvenpo — CHANGES_REQUESTED** [review]
You did not export the `crypto` subcrate from the main `secret_toolkit` namespace. I'll do it for you real quick

File: packages/crypto/src/sign.rs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+        let data = sha_256(b"test");
+        let signature = sign(&privkey, &data);
+
+        assert_eq!(
+            verify(&data, signature, &secp_pubkey.serialize_uncompressed()),
+            true
+        );
```
Comment (reuvenpo) [inline packages/crypto/src/sign.rs:60]: This can be simply:
```rust
assert!(verify(&data, signature, &secp_pubkey.serialize_uncompressed;
```

**Review by reuvenpo — APPROVED** [review]
awesome PR, thanks @PumpkinSeed :D

**Pull request merged**